### PR TITLE
Add groupby first and last aggregations

### PR DESCRIPTION
**Pull request by shwina**

### CODE REVIEW

**Comment by shwina** [comment]
@skirui-source @rjzamora I closed https://github.com/rapidsai/cudf/pull/8971 and opened this one after getting myself in a bit of a git mess. I addressed your review comments. Could you please take a look again? Apologies for the trouble.

**Review by skirui-source — APPROVED** [review]

**Review by cwharris — APPROVED** [review]

File: python/cudf/cudf/core/groupby/groupby.py [hunk 1]
```diff
@@ -817,6 +817,14 @@ def cummax(self):
         """Get the column-wise cumulative maximum value in each group."""
         return self.agg("cummax")
 
+    def first(self):
+        """Get the first non-null value in each group."""
+        return self.agg("first")
+
+    def last(self):
+        """Get the last non-null value in each group."""
+        return self.agg("last")
```
Comment (kkraus14) [inline python/cudf/cudf/core/groupby/groupby.py:826]: Should we add the `numeric_only` and `min_count` arguments just to be compatible with Pandas API?

**Review by rjzamora — APPROVED** [review]
Dask-related changes LGTM - Thanks Ashwin!

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**